libcodechecker/analyze/analysis_manager.py

```diff
@@ -192,6 +194,268 @@ def get_dependent_headers(buildaction, archive):
     return dependencies
 
 
+def __escape_source_path(source):
```

Comment:
This function output on `"   /home/   username   /projects/   simple/  main.cpp  "` will be the following: `" \ \ /home/\ \ \ username\ \ \ /projects/\ \ \ simple/\ \ main.cpp\ \ "`. So the leading space didn't escape.

I think this function can be written like so:
```python
return '\ '.join(source.replace('\ ', ' ').split(' '))
```
What do you think?

Reply:
I run the tests with the change and they pass. @martong what do you think? A separate unit test would be nice later for this escaping.

Reply:
Well, I prefer the more compact solution if that has the same functionality. Perhaps some unit tests of this function could be useful to verify if the compact version is the same.